### DISCUSSION TITLE
Upgrade using docker-compose.yml file

### DISCUSSION THREAD

**Discussion by stibra** [cybertec-postgresql/pg_timetable#554] [Q&A] [answered]
How to specify database schema upgrade switch in docker-compose.yml file?

**Comment by pashagolub** [comment]
Can you please provide the file you are using at the moment?

**Comment by stibra** [comment]
version: '3'
services:

scheduler:
    image: cybertecpostgresql/pg_timetable:latest
    restart: always
    environment:
      - PGTT_PGHOST=db
      - PGTT_CLIENTNAME=worker001
      - PGTT_PGPORT=5432
      - PGTT_PGUSER=postgresuser
      - PGTT_PGPASSWORD=password
      - PGTT_PGDATABASE=database_name
    depends_on:
      - db
    networks:
      - webproxy

**Comment by pashagolub** [comment, marked as answer]
can the `command: ["--upgrade"]` solve the issue?
https://docs.docker.com/compose/compose-file/05-services/#command

**UPD**:
It does work with docker-compose and docker swarm, but it does not work with podman play kube or kubernetes, command has to be like that:
`command: ["/pg_timetable", "--upgrade"]`

**Reply by stibra** [reply]
Thanks, it works now! I have tried with some older version before, there was some DSN string error, now it works like a charm.

**Reply by pashagolub** [reply]
You're more than welcome!

**Reply by stibra** [reply]
It does work with docker-compose and docker swarm, but it does not work with podman play kube or kubernetes, command has to be like that:
command: ["/pg_timetable", "--upgrade"]

**Reply by pashagolub** [reply]
Oh, strange enough. I would expect the opposite! :) But good to know we can still handle that

**Reply by stibra** [reply]
It is interesting that full command does not work with docker-compose and swarm.
